src/Display/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import Modal from '.';
+
+import * as renderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import { fireEvent, render } from '@testing-library/react';
+
+const props = {
+  title: 'Modal Title',
+  content: 'Modal Content',
+  footer: 'Modal Footer',
+  onClose: jest.fn(),
+};
+
+const OpenedModal = (
+  <Modal
+    title={props.title}
+    isVisible={true}
+    footer={[<button key="button">{props.footer}</button>]}
+    onClose={props.onClose}
+  >
+    <p>{props.content}</p>
+  </Modal>
+);
+
+function setupModal(isVisible: boolean) {
+  const onClose = jest.fn();
+  const ModalComponent = (
+    <Modal
+      title={props.title}
+      isVisible={isVisible}
+      footer={[<button key="button">{props.footer}</button>]}
+      onClose={onClose}
+    >
+      <p>{props.content}</p>
+    </Modal>
+  );
+  const { getByTestId, getByRole } = render(ModalComponent);
+  const modalContainer = getByTestId('modal-container');
+  const modalDialog = getByRole('dialog');
+  return [modalContainer, onClose, modalDialog];
+}
+
+it('<Modal> should render with a title, content, footer and an onClick handler', () => {
+  const ModalSnapshot = renderer.create(OpenedModal).toJSON();
+  expect(ModalSnapshot).toMatchSnapshot();
+});
+
+describe('when modal is opened', () => {
+  it('should be visible', () => {
+    const [modalContainer] = setupModal(true);
+    expect(modalContainer).toHaveStyle('visibility: visible');
+  });
+
+  it('should not be closed if the modal is clicked', () => {
+    const [, onClose, modalDialog] = setupModal(true);
+    fireEvent.click(modalDialog as Element);
+    expect(onClose).toHaveBeenCalledTimes(0);
+  });
+});
+
+describe('when modal is opened', () => {
```

Comment:
Hey @franciscrispin, I have a question.

The group of tests here are checking whether or not the title and content and footer is present in accordance to the props.

However, at `Line 44`, there is a snapshot check. 

Is there a reason why this is repeated?
Also, which is a healthier way of testing?

Reply:
Oh hmm I think snapshot testing at the component level is to ensure that the component as a whole does not change unexpectedly. 

From the docs https://jestjs.io/docs/en/snapshot-testing
```
Snapshot tests are a very useful tool whenever you want to make sure your UI does not change unexpectedly.
```

Whereas the unit tests here are to test a specific part of the component. In a sense it is repeated but the purpose of the snapshot test is not to test each specific part, but more of testing the component as a whole.